src/nvim/drawline.c

```diff
@@ -395,6 +398,93 @@ static int get_sign_attrs(buf_T *buf, linenr_T lnum, SignTextAttrs *sattrs, int
   return num_signs;
 }
 
+static void get_statuscol_str(win_T *wp, linenr_T lnum, int row, int startrow, int filler_lines,
+                              int cul_attr, int sign_num_attr, SignTextAttrs *sattrs,
+                              foldinfo_T foldinfo, char_u *extra, statuscol_T *stcp)
+{
+  int sign_idx = 0;
+  // Set number, fold and sign text and attrs when not wrapped
+  if (row == startrow + filler_lines) {
+    set_vim_var_nr(VV_LNUM, lnum);
+    set_vim_var_nr(VV_RELNUM, labs((long)get_cursor_rel_lnum(wp, lnum)));
+    stcp->num_attr = sign_num_attr ? sign_num_attr
+                     : get_line_number_attr(wp, lnum, row, startrow, filler_lines);
+
+    if (compute_foldcolumn(wp, 0)) {
+      stcp->fold_attr = win_hl_attr(wp, use_cursor_line_sign(wp, lnum) ? HLF_CLF : HLF_FC);
+      size_t n = fill_foldcolumn((char_u *)stcp->fold_text, wp, foldinfo, lnum);
+      stcp->fold_text[n] = NUL;
+    }
+
+    for (; sign_idx < wp->w_scwidth; sign_idx++) {
+      SignTextAttrs *sattr = sign_get_attr(sign_idx, sattrs, wp->w_scwidth);
+
+      stcp->sign_text[sign_idx] = (sattr && sattr->text) ? sattr->text : "  ";
+      stcp->sign_attr[sign_idx] = (use_cursor_line_sign(wp, lnum) && cul_attr > 0) ?
+                                  cul_attr : sattr ? sattr->hl_attr_id : 0;
+    }
+  }
+  stcp->sign_text[sign_idx] = NULL;
+
+  StlClickRecord *clickrec;
+
+  set_vim_var_bool(VV_WRAP, row != startrow + filler_lines);
+  int width = build_stl_str_hl(wp, stcp->text, MAXPATHL, wp->w_p_stc, "statuscolumn",
+                               OPT_LOCAL, ' ', stcp->width, &stcp->hlrec, &clickrec, stcp);
+
+  // Force a redraw in case of error or when truncated
+  if (*wp->w_p_stc == NUL || (stcp->truncate > 0 && wp->w_nrwidth < 19)) {
+    if (stcp->truncate) {  // Avoid truncating 'statuscolumn'
+      wp->w_nrwidth_width = MIN(19, wp->w_nrwidth_width + stcp->truncate);
+    } else {  // 'statuscolumn' reset due to error
+      wp->w_nrwidth_line_count = 0;
+    }
+    wp->w_redr_statuscol = true;
+    return;
+  }
+  // Reset text pointer and current attr for new line
+  stcp->textp = stcp->text;
+  stcp->cur_attr = stcp->num_attr;
+  stcp->text_len = STRLEN(stcp->text);
```

Comment:
On Mac OS, I cannot build unless I make "STRLEN" lowercase.

There are three 3 instances of STRLEN. 
src/nvim/drawline.c:448
src/nvim/drawline.c:476
src/nvim/statusline.c:1580

Reply:
Thanks, `STRLEN` was removed in https://github.com/neovim/neovim/commit/019c8805e514428c0b999583f5aec8c9f4eb96d0. I guess you're applying this PR on top of master?

Reply:
> Thanks, `STRLEN` was removed in [019c880](https://github.com/neovim/neovim/commit/019c8805e514428c0b999583f5aec8c9f4eb96d0). I guess you're applying this PR on top of master?

Yup, I am rebasing on master.